server/protocols/jmap-rfc-8621/src/test/scala/org/apache/james/jmap/http/BasicAuthenticationStrategyTest.scala

```diff
@@ -57,25 +56,67 @@ class BasicAuthenticationStrategyTest {
   @BeforeEach
   def setup(): Unit = {
     when(mockedRequest.requestHeaders).thenReturn(mockedHeaders)
-    testee = new BasicAuthenticationStrategy(basicAuthenticationTokenManager, mailboxManager)
+    testee = new BasicAuthenticationStrategy(usersRepository, mailboxManager)
   }
 
   @Test
-  def shouldReturnAnyUsernameWhenValidBasicAuthToken(): Unit = {
+  def shouldReturnSessionWhenValidTokenAndExistedUser(): Unit = {
     Mockito.when(mockedRequest.requestHeaders().getAll(AUTHORIZATION_HEADERS)).thenReturn(ImmutableList.of(userExistedToken))
-    Assertions.assertEquals(testee.createMailboxSession(mockedRequest).block.getUser, username1)
+    Assertions.assertEquals(testee.createMailboxSession(mockedRequest).block.getUser, username)
   }
 
   @Test
-  def shouldThrowWhenInvalidBasicAuthToken(): Unit = {
-    Mockito.when(mockedHeaders.getAll(AUTHORIZATION_HEADERS)).thenReturn(ImmutableList.of("invalid"))
+  def shouldThrowWhenInvalidToken(): Unit = {
+    Mockito.when(mockedHeaders.getAll(AUTHORIZATION_HEADERS)).thenReturn(ImmutableList.of(tokenWithInvalidCredential))
+    Assertions.assertThrows(classOf[IllegalArgumentException], () => testee.createMailboxSession(mockedRequest).block())
+  }
+
+  @Test
+  def shouldThrowWhenTokenWithOutUsername(): Unit = {
+    Mockito.when(mockedHeaders.getAll(AUTHORIZATION_HEADERS)).thenReturn(ImmutableList.of(emptyUserNameToken))
```

Comment:
Please inline these `emptyUserNameToken` variables, having to navigate between here and the feature makes it hard to read your work.

Reply:
Why don't we just test parsing on `UserCredentialParse::parseUserCredentials` ? 

We won't need mockito anymore.... (the granularity of your test is not small enough)